feat(segmentation): Enhanced segmentation panel design for TMTV

IbrahimCSAE opened the pull request:
### Context

Addresses issue #3790 

### Changes & Results

Pending

### Screenshots

![image](https://github.com/OHIF/Viewers/assets/93064150/95a7ac63-ab57-4828-a399-653a2aef5cfb)

![image](https://github.com/OHIF/Viewers/assets/93064150/ee9fd650-dcb6-450b-8ddf-943f6343cca5)

![image](https://github.com/OHIF/Viewers/assets/93064150/c37748bd-8a8c-4bcd-90b1-de77cd6a4460)

![image](https://github.com/OHIF/Viewers/assets/93064150/a58c6464-3c18-46d8-9666-c3a25b729a2e)

![image](https://github.com/OHIF/Viewers/assets/93064150/ff9a6b95-609e-4a1a-9f5f-0d9cfafbf7c6)


### Notes

Most work is complete, remaining piece is to switch out the TMTV panel

sedghi commented on platform/ui/src/assets/styles/styles.css:
```diff
@@ -1,7 +1,7 @@
 .ohif-scrollbar {
   scrollbar-color: #173239 transparent;
-  scrollbar-gutter: stable;
+  overflow-y: auto;
```
i'm not sure why you are removing the gutter, it is needed to overlay the thumnail on top of the div, no?

sedghi commented on extensions/cornerstone-dicom-seg/src/panels/PanelSegmentation.tsx:
```diff
@@ -224,54 +224,105 @@ export default function PanelSegmentation({
-            _setSegmentationConfiguration(selectedSegmentationId, 'fillAlphaInactive', value)
-          }
-        />
+        {configuration.segmentationPanelMode === 'dropdown' ? (
```
this looks redundant, can you not do it here, and pass the mode inside ui and do it there?

IbrahimCSAE replied:
I did something different, let me know if that's fine

sedghi commented on platform/ui/src/components/SegmentationGroupTable/SegmentationDropDownRow.tsx:
```diff
@@ -122,8 +122,24 @@ function SegmentationDropDownRow({
+          <Icon
+            name="info-action"
+            style={{
+              color: '#348CFD',
```
please don't hard code the color, if we need a new color add it to tailwind config

sedghi reviewed (commented):
see my comments please, great work

sedghi reviewed (commented):
can we make the scrollbar outside the header 

<img width="265" alt="CleanShot 2024-03-27 at 12 22 50@2x" src="https://github.com/OHIF/Viewers/assets/7490180/b1871aee-8cf3-47ae-be15-e56e8a06b45d">

----



number box is too big
<img width="226" alt="CleanShot 2024-03-27 at 12 23 05@2x" src="https://github.com/OHIF/Viewers/assets/7490180/ce32ba0a-c0ef-42f8-922e-20f08d30f5eb">


---

needs to be half 

<img width="220" alt="CleanShot 2024-03-27 at 12 23 28@2x" src="https://github.com/OHIF/Viewers/assets/7490180/aff9780c-95b4-44c2-9403-444754b54c46">


---

can you add the new segmetnation to the tmtv 

<img width="257" alt="CleanShot 2024-03-27 at 12 24 05@2x" src="https://github.com/OHIF/Viewers/assets/7490180/3807d189-bd05-40b5-8a7d-79700360b657">

IbrahimCSAE commented:
Fixes:

1. Scrollbar outside header for dropdown and expanded (I needed to set max height 500px]

![image](https://github.com/OHIF/Viewers/assets/93064150/ef9f46de-06b3-4f53-8ff8-c55ce35f0ee9)

![image](https://github.com/OHIF/Viewers/assets/93064150/56e78325-f910-480d-b4c4-4dbbce7b5fb4)

2.number box sizing fixed

![image](https://github.com/OHIF/Viewers/assets/93064150/b186ae52-ef32-446b-a07f-69431c5282b1)

3. Reload data patient info size is corrected and smaller

![image](https://github.com/OHIF/Viewers/assets/93064150/61ba8146-687e-43d1-bdde-a93d6265b22a)

4. Added new segmentation panel to tmtv (some features do not work as I'm not sure how it should function in tmtv)

![image](https://github.com/OHIF/Viewers/assets/93064150/adc7b409-b874-4d67-a30d-bdc348ba0970)

5. Fixed visibility toggle bug, now it works as expected

6. fixed I icon location when expanded/unexpanded

![image](https://github.com/OHIF/Viewers/assets/93064150/4913c667-3ef2-47db-a47d-a92860ca4f46)

7. scroll bars start after header now

8.Fixed add segment padding

![image](https://github.com/OHIF/Viewers/assets/93064150/68a4b819-e5d1-4f5a-bfb5-d2e682497b7e)

9. working on it

10. centered settings icon

![image](https://github.com/OHIF/Viewers/assets/93064150/a08a00a9-04e1-4bc9-b08c-252237e73a29)

11. Fixed add segmentation padding

![image](https://github.com/OHIF/Viewers/assets/93064150/8ae78017-c064-4321-ae5b-7624fac225dd)


@sedghi please take a look, we just need to add the tools and fix up panel in tmtv.

IbrahimCSAE commented on extensions/default/src/Toolbar/Toolbar.tsx:
```diff
@@ -33,16 +31,7 @@ export function Toolbar({ servicesManager }) {
           />
         );
 
-        return disabled ? (
```
duplicate tooltip bug

IbrahimCSAE commented on platform/ui/src/components/Tooltip/Tooltip.tsx:
```diff
@@ -67,37 +154,7 @@ const Tooltip = ({
-          </svg>
-        </div>
-      </div>
+      {tooltipContainer && ReactDOM.createPortal(tooltipContent, tooltipContainer)}
```
react portals

IbrahimCSAE commented on modes/segmentation/src/segmentationButtons.ts:
```diff
@@ -95,7 +95,7 @@ const toolbarButtons: Button[] = [
           id: 'Threshold',
           icon: 'icon-tool-threshold',
-          label: 'Eraser',
+          label: 'Threshold Tool',
```
fix tool name

IbrahimCSAE commented:
![image](https://github.com/OHIF/Viewers/assets/93064150/55fc4e29-a037-49c0-b2a0-6caa71c09e95)

react portals in action, looks good

sedghi commented on platform/ui/src/components/Tooltip/Tooltip.tsx:
```diff
@@ -57,8 +62,90 @@ const Tooltip = ({
 
   const isOpen = (isSticky || isActive) && !isDisabled;
 
+  useEffect(() => {
+    if (isOpen && parentRef.current && tooltipRef.current) {
```
Nice! 

Can you please add a comment on what is happening for future devs, newX, newY, parent ...

sedghi replied:
if (!isOpen || !parentRef.current || !tooltipRef.current){
  return
}

--> less indentation --> more clear

sedghi reviewed (commented):
Getting there, 

Patient info has wrong icon 

<img width="183" alt="CleanShot 2024-04-02 at 09 10 35@2x" src="https://github.com/OHIF/Viewers/assets/7490180/d72eeb1a-c106-48d1-8149-f44ac99b0fa2">


---------
The dropdown for the modal seems like getting in the div (maybe we need portal here as well?)

https://github.com/OHIF/Viewers/assets/7490180/b47619bb-2440-4414-b275-a29b9177db1f


--------

Empty Segmentation should have one segment at least, so we should add segment 1 programmatically

https://github.com/OHIF/Viewers/assets/7490180/3287ae87-4432-4ea8-aadc-77435960ccbf


----

suv peak display text does not appear for some reason below the segment index


https://github.com/OHIF/Viewers/assets/7490180/197f6fc4-50d4-4fd5-86c8-5bb94a7a11eb

----

seems like active segment cannot be ch

https://github.com/OHIF/Viewers/assets/7490180/58535523-b804-4331-9f79-41be172d9c77

anged
